### ISSUE TITLE
Phase 2: Text classification transform

### ISSUE THREAD

**Issue by luisquintanilla** [luisquintanilla/mlnet-text-inference-custom-transforms#3]
## Overview

Add a text classification post-processing transform that takes raw ONNX logits and produces class probabilities + predicted labels. This is the simplest new task  no tokenizer or scorer changes required.

## Architecture

```
TextTokenizerTransformer  OnnxTextModelScorerTransformer  SoftmaxClassificationTransformer
                                                                        
                                                            
                                                                                   
                                                    Probabilities[]          PredictedLabel
```

## What to implement

### 1. `SoftmaxClassificationOptions` (config class)

```csharp
public class SoftmaxClassificationOptions
{
    /// <summary>Name of the input column containing raw logits. Default: "RawOutput".</summary>
    public string InputColumnName { get; set; } = "RawOutput";

    /// <summary>Name of the output column for class probabilities. Default: "Probabilities".</summary>
    public string ProbabilitiesColumnName { get; set; } = "Probabilities";

    /// <summary>Name of the output column for the predicted label. Default: "PredictedLabel".</summary>
    public string PredictedLabelColumnName { get; set; } = "PredictedLabel";

    /// <summary>
    /// Label names corresponding to each output class index.
    /// If null, predicted label will be the integer class index as a string.
    /// </summary>
    public string[]? Labels { get; set; }

    /// <summary>Number of output classes. Auto-detected from model output if not set.</summary>
    public int? NumClasses { get; set; }
}
```

### 2. `SoftmaxClassificationEstimator : IEstimator<SoftmaxClassificationTransformer>`

**`Fit()` should:**
- Validate that `InputColumnName` exists in the input schema
- If `Labels` is provided and `NumClasses` is null, set `NumClasses = Labels.Length`
- Create the transformer

**`GetOutputSchema()` should:**
- Add `ProbabilitiesColumnName` as `Vector<Single, NumClasses>`
- Add `PredictedLabelColumnName` as `Text`

### 3. `SoftmaxClassificationTransformer : ITransformer`

**`Transform()`:** Return a wrapping `ClassificationDataView` (lazy  no computation here)

**Internal direct face: `Classify(float[][] rawOutputs)  ClassificationResult[]`**

**Cursor logic (in `ClassificationCursor`):**
1. Read `RawOutput` (float[] of length numClasses)
2. Apply softmax: use `TensorPrimitives.SoftMax(logits, probabilities)` from `System.Numerics.Tensors`
3. Argmax: `int predictedIndex = TensorPrimitives.IndexOfMax(probabilities)`
4. Map to label: `predictedLabel = Labels?[predictedIndex] ?? predictedIndex.ToString()`
5. Serve `Probabilities` via `VBuffer<float>` getter
6. Serve `PredictedLabel` via `ReadOnlyMemory<char>` getter

### 4. `OnnxTextClassificationEstimator` (facade)

Chains: `TextTokenizerEstimator`  `OnnxTextModelScorerEstimator`  `SoftmaxClassificationEstimator`

Follow the exact same pattern as `OnnxTextEmbeddingEstimator.Fit()`:
1. Create and fit the tokenizer
2. Create and fit the scorer (with `PreferredOutputNames = ["logits", "output"]`)
3. Auto-detect `NumClasses` from scorer output dimension
4. Create and fit the classification post-processor

### 5. Extension methods (add to `MLContextExtensions.cs`)

```csharp
public static SoftmaxClassificationEstimator SoftmaxClassify(
    this TransformsCatalog catalog, SoftmaxClassificationOptions options)

public static OnnxTextClassificationEstimator OnnxTextClassification(
    this TransformsCatalog catalog, OnnxTextClassificationOptions options)
```

### 6. File layout

```
Classification/
 SoftmaxClassificationOptions.cs
 SoftmaxClassificationEstimator.cs
 SoftmaxClassificationTransformer.cs    (includes ClassificationDataView + ClassificationCursor)
 OnnxTextClassificationOptions.cs
 OnnxTextClassificationEstimator.cs     (facade)
```

## Samples

### `samples/Classification/SentimentDistilBERT/`
- **Model**: `distilbert-base-uncased-finetuned-sst-2-english`
- **Architecture**: DistilBERT (no token_type_ids)
- **Labels**: `["NEGATIVE", "POSITIVE"]`
- **Shows**: Basic binary sentiment classification

```powershell
Invoke-WebRequest -Uri "https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english/resolve/main/onnx/model.onnx" -OutFile "models/model.onnx"
Invoke-WebRequest -Uri "https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english/resolve/main/vocab.txt" -OutFile "models/vocab.txt"
```

### `samples/Classification/EmotionRoBERTa/`
- **Model**: `SamLowe/roberta-base-go_emotions`
- **Architecture**: RoBERTa (BPE tokenizer)
- **Labels**: 28 emotion labels
- **Shows**: Multi-class with many labels, BPE tokenizer auto-detection

### `samples/Classification/ZeroShotDeBERTa/`
- **Model**: `MoritzLaurer/DeBERTa-v3-base-mnli-fever-anli`
- **Architecture**: DeBERTa-v3
- **Labels**: `["contradiction", "neutral", "entailment"]`
- **Shows**: NLI-based zero-shot classification

Each sample needs: `Program.cs`, `README.md` (with download commands + expected output), `models/.gitkeep`, `.csproj`

## Verification

- [ ] `dotnet build` passes
- [ ] SentimentDistilBERT sample produces correct sentiment predictions
- [ ] EmotionRoBERTa sample runs with 28-class output
- [ ] ZeroShotDeBERTa sample runs with NLI labels
- [ ] Composable pipeline (tokenize  score  classify) matches facade output
- [ ] Chained estimator pipeline (`.Append()`) works
- [ ] Existing embedding samples still work (no regression)

## Dependencies
- Requires Phase 1 (issue #2)
- Classification models output `[batch, num_classes]` (rank 2)  the scorer already handles rank-2 via `HasPooledOutput=true`. The classification post-processor interprets those values as logits.

## Key implementation note
The classification output tensor is rank-2 `[batch, num_classes]`, same shape as pre-pooled embeddings. The scorer's `HasPooledOutput` flag will be true. The classification post-processor should NOT use this flag  it should just read the raw output and apply softmax regardless.
